Microsoft.Toolkit.Uwp.UI.Animations/Xaml/AnimationDictionary.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations.Xaml
+{
+    /// <summary>
+    /// A collection of animations that can be defined from XAML.
+    /// </summary>
+    public sealed class AnimationDictionary : DependencyObject, IList<AnimationCollection2>
```

Comment:
This almost seems pretty similar to the current AnimationCollection we have in the toolkit...? 🤔

Reply:
Yeah the two types share some similarities indeed 🙂

The main difference is that `AnimationDictionary` only acts as a transparent container for animations, but it itself doesn't contain any logic to create or start animations. It's only responsible for binding contained animations to the parent `UIElement` it's attached to, while the actual animation logic is decoupled and only defined within each actual animation that can be executed. Whereas `AnimationCollection` is itself an animation that can be triggered, and that contains a number of individual animations. The key point is that where previously we only had the `Implicit` types, those could only accept a single `AnimationCollection`, since there's only a single animation group you can assign to an implicit animation set for an item. Whereas now with explicit animations too, you're free to define as many different animation collections as you want for an item, and trigger them individually. Basically:

- `AnimationDictionary`: a container for composite animations attached to a control.
- `AnimationCollection`: a single composite animation (like a `Storyboard`) that can be triggered
- `ITimeline`: a single animation or scope within a collection (like a `DoubleAnimation`)

This also gives us flexibility to still let users define a single collection when assigning to an implicit animation, or to define an arbitrary number of animations when defining explicit animations for controls.